### PR TITLE
Adding the convert scripts which will now prevent converting models

### PR DESCRIPTION
**Pull request by Narsil**
in case they will trigger warnigns in the `transformers` side.
Even if the model is perfectly fine, core maintainers fear an influx
of opened issues.

This is perfectly legit.
On the `transformers` side fixes are on the way: https://github.com/huggingface/transformers/pull/20042

We can wait for this PR to hit `main` before communicating super widely.

In the meantime this script of convertion will now prevent converting
models that would trigger such warnings (so the output of the script
**will** depend on the `transformers` freshness.

### CODE REVIEW

File: bindings/python/convert.py [hunk 1]
```diff
@@ -97,16 +103,17 @@ def convert_single(model_id: str, folder: str) -> List["CommitOperationAdd"]:
...
 
-    _, pt_model = infer_framework_load_model(model_id, config)
-    _, sf_model = infer_framework_load_model(folder, config)
+    _, (pt_model, pt_infos) = infer_framework_load_model(model_id, config, output_loading_info=True)
+    _, (sf_model, sf_infos) = infer_framework_load_model(folder, config, output_loading_info=True)
 
-    pt_model = pt_model
-    sf_model = sf_model
+    if pt_infos != sf_infos:
+        raise ValueError(f"different infos {model_id}")
```
Comment (sgugger) [inline bindings/python/convert.py:116]: Maybe add some info on the differences?

File: bindings/python/convert.py [hunk 2]
```diff
@@ -156,7 +163,7 @@ def previous_pr(api: "HfApi", model_id: str, pr_title: str) -> Optional["Discuss
             return discussion
 
 
-def convert(api: "HfApi", model_id: str, force: bool=False) -> Optional["CommitInfo"]:
+def convert(api: "HfApi", model_id: str, force: bool = False) -> Optional["CommitInfo"]:
```
Comment (sgugger) [inline bindings/python/convert.py:166]: Didn't catch this before but we shouldn't use any api object and just the top level methods (`create_commit` is directly in the main init).

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

File: bindings/python/convert.py (hunk 1 above)
Comment (Narsil) [reply bindings/python/convert.py:116]: It's quite noisy which is why I didn't include it (I mean infos can be quite numerous even on pure PT for transformers).

Doing a proper diff could help mitigate that but also relatively verbose code. I'll give it a stab.

File: bindings/python/convert.py (hunk 2 above)
Comment (Narsil) [reply bindings/python/convert.py:166]: Why ?

I figured it was actually better to pass around an HfApi codewise, so you can include a specific token (for instance to create conversions directly).
The original space convertion was passing around a Token, but I figure `HfApi` was even easier to transport around and there's less risk to forget to include `token=token` in whatever call.

Comment (sgugger) [reply bindings/python/convert.py:166]: I don't think anyone who uses hugginface_hub uses the HfApi object.

Comment (sgugger) [reply bindings/python/convert.py:166]: You should include a kwarg for token instead IMO

Comment (julien-c) [reply bindings/python/convert.py:166]: > I don't think anyone who uses hugginface_hub uses the HfApi object.

I do  :)

Comment (julien-c) [reply bindings/python/convert.py:166]: (this feature of setting a token on HfApi is quite recent though, but it's useful IMO)

Comment (sgugger) [reply bindings/python/convert.py:166]: Let me rephrase, I don't think many people except Julien use `HfApi` :-p

Comment (Narsil) [reply bindings/python/convert.py:166]: I did use it naturally actually an I think it's pretty cool to encapsulate common client options (most important of which the token).

I really think the "oh but you forgot to pass the token" is a real failure case, and it did happen when I used `get_repo_discussions` where I failed to pass the token and it would fail on private models on the space.

File: bindings/python/convert.py (hunk 1 above)
Comment (Narsil) [reply bindings/python/convert.py]: Diff added.

File: bindings/python/convert.py (hunk 2 above)
Comment (Narsil) [reply bindings/python/convert.py:183]: I'll go ahead and merge this.

Please open an issue if you feel so strongly about the `HfApi` object :)
